# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGAnimatedLength.cpp
# Commit: 5158484e9eab
# Full Hash: 5158484e9eabc55a48e73e68ad4b6e676d057997
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 35034282dc7c -r 5158484e9eab dom/svg/SVGAnimatedLength.cpp
--- a/dom/svg/SVGAnimatedLength.cpp	Mon Sep 11 18:20:18 2023 +0000
+++ b/dom/svg/SVGAnimatedLength.cpp	Mon Sep 11 18:20:19 2023 +0000
@@ -178,6 +178,11 @@
   return GetEmLength(aType);
 }
 
+CSSSize UserSpaceMetrics::GetCSSViewportSizeFromContext(
+    const nsPresContext* aContext) {
+  return CSSPixel::FromAppUnits(aContext->GetSizeForViewportUnits());
+}
+
 SVGElementMetrics::SVGElementMetrics(const SVGElement* aSVGElement,
                                      const SVGViewportElement* aCtx)
     : mSVGElement(aSVGElement), mCtx(aCtx) {}
@@ -207,6 +212,17 @@
   return FixAxisLength(mCtx->GetLength(aCtxType));
 }
 
+CSSSize SVGElementMetrics::GetCSSViewportSize() const {
+  if (!mSVGElement) {
+    return {0.0f, 0.0f};
+  }
+  nsPresContext* context = nsContentUtils::GetContextForContent(mSVGElement);
+  if (!context) {
+    return {0.0f, 0.0f};
+  }
+  return GetCSSViewportSizeFromContext(context);
+}
+
 bool SVGElementMetrics::EnsureCtx() const {
   if (!mCtx && mSVGElement) {
     mCtx = mSVGElement->GetCtx();
@@ -267,6 +283,10 @@
   return SVGIntegrationUtils::GetSVGCoordContextForNonSVGFrame(mFrame);
 }
 
+CSSSize NonSVGFrameUserSpaceMetrics::GetCSSViewportSize() const {
+  return GetCSSViewportSizeFromContext(mFrame->PresContext());
+}
+
 float UserSpaceMetricsWithSize::GetAxisLength(uint8_t aCtxType) const {
   gfx::Size size = GetSize();
   float length;